vmodel modifiers clarify example and diff 2.x v 3.x (#344)
vuejs/docs-next: src/guide/component-custom-events.md

## Handling `v-model` modifiers

> When we were learning about form input bindings, we saw that `v-model` has [built-in modifiers](/guide/forms.html#modifiers) like `.trim` or `.native`. In some cases, however, you might also want to add your own custom modifiers.

Reviewer: My bad, I haven't caught it on the first round of review: `.native` does not exist anymore, so we should remove it

Author: Oops, yeah. Removed.

> In Vue 2.x, we had hard-coded support for modifiers like `.trim` or `.native` on component `v-model` bindings. And even though Vue 3.x also supports [out of the box modifiers](/guide/forms.html#modifiers), it would be more useful if our components could support custom modifiers of our own. 

Reviewer: **suggestion**: I think we should remove 2.x mention here completely and just mention we have built-in modifiers with the link as nothing changes except `.native` modifier.

Also, let's remove the statement about custom modifiers being **more** useful (it really depends on the use-case). What about just `useful`?

```suggestion
As we already mentioned, `v-model` has [built-in modifiers](/guide/forms.html#modifiers) like `.trim` or `.native`. In some cases, you might also want to add your own custom modifier.
```

Let's create an example custom modifier, `capitalize`, that capitalizes the first letter of the string provided by the `v-model` binding.

> In 3.x, modifiers added to a component `v-model` will be provided to the component via the `modelModifiers` prop. In the below example, we have created a component that contains a `modelModifiers` prop and defaults to an empty object.

Reviewer: **suggestion**: we are on 3.x docs already so we could remove version mention 😅 

```suggestion
Modifiers added to a component `v-model` will be provided to the component via the `modelModifiers` prop. In the below example, we have created a component that contains a `modelModifiers` prop that defaults to an empty object.
```

> Notice that when the component's `created` lifecycle hook triggers, the `modelModifiers` prop contains `capitalize` and it's value is `true` - due to it being set on the `v-model` binding `v-model.capitalize="bar"`.

Reviewer: ```suggestion
Notice that when the component's `created` lifecycle hook triggers, the `modelModifiers` prop contains `capitalize` and its value is `true` - due to it being set on the `v-model` binding `v-model.capitalize="bar"`.
```

Author: oops 😅
